src/components/Countdown.tsx

```diff
@@ -0,0 +1,93 @@
+import React, { FC, useState } from 'react';
+import { Text, StyleSheet, View } from 'react-native';
+
+import { DateType } from 'app/consts';
+import { getTimeDiff, getTimeDuration } from 'app/helpers/date';
+import { useInterval } from 'app/helpers/useInterval';
+import { palette, fonts } from 'app/styles';
+
+const i18n = require('../../loc');
+
+interface CellsProps {
+  text: string;
+  title: string;
+}
+
+const Cells: FC<CellsProps> = ({ text, title }) => (
+  <View style={styles.cellsWrapper}>
+    <View style={styles.lettersWrapper}>
+      {text.split('').map((letter, index) => (
+        <View key={index} style={styles.cell}>
+          <Text style={styles.cellText}>{letter}</Text>
+        </View>
+      ))}
+    </View>
+    <Text style={styles.cellsTitle}>{title}</Text>
+  </View>
+);
+
+interface Props {
+  dataEnd: DateType;
+}
+
+const numberToCellString = (n: number) => n.toString().padStart(2, '0');
+
+export const Countdown: FC<Props> = ({ dataEnd }) => {
+  const getCurrentTime = () => getTimeDiff(new Date(), dataEnd);
+
+  const [time, setTime] = useState(getCurrentTime());
+
+  useInterval(
+    () => {
+      setTime(getCurrentTime());
+    },
+    time > 0 ? 1000 : null,
+  );
```

Comment:
Are you sure it's working correctly? If I'm not wrong, in this case the component is re-rendered every second and eventually we have multiple `useInterval()` called at the same time. I'd suggest wrapping it with `useEffect()`

Reply:
Good point, I assumed it s handled inside the hook, but worth checking for sure

Reply:
I checked and it is firing only once